kola/tests/coretest/fleet.go

```diff
@@ -73,37 +35,39 @@ func TestFleetctlRunService() error {
 		return fmt.Errorf("Failed writing %v: %v", serviceFile.Name(), err)
 	}
 
-	defer timeoutFleetctl("destroy", serviceFile.Name())
+	myid := MachineID()
+
+	fleetChecker := func() error {
+		stdout, stderr, err := Run("fleetctl", "list-machines", "-no-legend", "-l", "-fields", "machine")
+		if err != nil {
+			return fmt.Errorf("fleetctl list-machines failed with error: %v\nstdout: %s\nstderr: %s", err, stdout, stderr)
+		}
+
+		if !strings.Contains(stdout, myid) {
+			return fmt.Errorf("fleetctl list-machines: machine ID %q missing from output\nstdout: %v\nstderr: %v", myid, stdout, stderr)
+		}
+
+		return nil
+	}
+
+	if err := util.Retry(5, 5*time.Second, fleetChecker); err != nil {
+		return err
+	}
 
-	stdout, stderr, err := timeoutFleetctl("start", serviceFile.Name())
+	stdout, stderr, err := Run("fleetctl", "start", "-block-attempts", "20", serviceFile.Name())
 	if err != nil {
 		return fmt.Errorf("fleetctl start failed with error: %v\nstdout: %s\nstderr: %s", err, stdout, stderr)
 	}
 
-	stdout, stderr, err = timeoutFleetctl("unload", serviceName)
+	stdout, stderr, err = Run("fleetctl", "unload", "-block-attempts", "20", serviceName)
 	if err != nil {
 		return fmt.Errorf("fleetctl unload failed with error: %v\nstdout: %s\nstderr: %s", err, stdout, stderr)
 	}
 
-	stdout, stderr, err = timeoutFleetctl("destroy", serviceName)
+	stdout, stderr, err = Run("fleetctl", "destroy", serviceName)
 	if err != nil {
 		return fmt.Errorf("fleetctl destroy failed with error: %v\nstdout: %s\nstderr: %s", err, stdout, stderr)
 	}
-	return nil
-}
 
-func timeoutFleetctl(action string, unitName string) (stdout string, stderr string, err error) {
```

Comment:
it already leaks processes on the system if it timed out anyway, since we never send kill if the timer expires...